채널 추가 페이지

hyerindev opened the pull request:
## 🔎 What is this PR?

### Figma: 프로토타입 캡처

<table>
  <tr>
    <th>모임 추가하기 페이지<br />첫 화면</th>
    <th>모임 추가하기 페이지<br />문자 선택, 입력 전</th>
    <th>모임 추가하기 페이지<br />문자 선택, 입력 후</th>
    <th>모임 추가하기 페이지<br />등반 선택</th>
  </tr>
  <tr>
    <td valign="top">
      <img src="https://github.com/user-attachments/assets/47764c6a-f155-486f-b928-c7d014537993" />
    </td>
    <td valign="top">
      <img src="https://github.com/user-attachments/assets/2831359d-4e7c-477c-9c14-e2d4915c50e5" />
    </td>
    <td valign="top">
      <img src="https://github.com/user-attachments/assets/55be4cde-684d-4089-b4ae-324ea300dec7" />
    </td>
    <td valign="top">
      <img src="https://github.com/user-attachments/assets/eca72c8d-a089-4bae-b22a-378de7192cc2" />
    </td>
  </tr>
</table>

## ✨ 설명

- 코드 재활용성을 높이고자, 페이지에 사용된 모든 인풋필드를 확장성을 고려하여 공통 컴포넌트로 개발했습니다.
  - #44 : type prop의 `'short'` / `'long'` 값에 따라 태그 및 스타일이 달라집니다. limit prop로 최대 입력 글자 수 제한 및 글자 수 표시를 on/off할 수 있습니다.
  - #45 : 라디오버튼 목록을 데이터화하여 넘기면 끝! (@rwaeng 감사합니다😊)
    CSS 선택자를 적극 활용하여 코드 분량을 확! 줄였습니다.
  - #46 : type prop의 `'date'` / `'period'` 값에 따라 날짜를 고를지 기간을 고를지 달라집니다. `'period'`로 지정한 경우, start 인풋필드는 end 인풋필드에서 입력한 날짜보다 이후의 날짜를 선택할 수 없도록 로직을 구성했습니다. (반대의 경우도 그렇습니다.)
  - #47 : 선택지를 데이터화하여 넘기면 스타일을 자동으로 입혀줍니다. select 태그는 placeholder 속성을 갖지 않지만, 자체적으로 구현했습니다. placeholder용 option 태그는 disabled 속성을 주어 선택할 수 없도록 신경썼습니다.
  - #48 : legend를 표시하고 인풋필드를 그룹화하는 역할입니다.
- `path` 태그 `fill` 속성의 `currentColor` 키워드를 활용하여, 일일히 제어할 필요 없이 부모의 color에 따라 fill이 바뀌도록 했습니다. [레퍼런스](https://ddd120.tistory.com/60)
- submit [버튼에 form 속성](https://developer.mozilla.org/ko/docs/Web/HTML/Element/button#form)을 추가하여 form과 연결시켰습니다.
- [select 태그](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select)와 [option 태그](https://developer.mozilla.org/ko/docs/Web/HTML/Element/option)를 활용하여 InputDropdown 컴포넌트 (공통)을 개발했습니다.
- `<input type="date">`를 지원하지 않는 브라우저를 위해 pattern 속성을 지정했습니다. [레퍼런스](https://developer.mozilla.org/ko/docs/Web/HTML/Element/input/date#미지원_브라우저_처리하기)
- [toLocaleDateString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) 함수와 정규식을 활용하여 ChannelAddPage에서 InputDatepicker로 min props를 넘길 때 데이터를 쉽게 포맷팅했습니다.
- '등반 기간' 항목은 기간 형태로 입력받도록 설정했습니다. 추후 디자인이 확정되면 수정하겠습니다.

## 📷 스크린샷 (선택)

<table>
  <tr>
    <th>문자 모임 추가하기</th>
    <th>전화 모임 추가하기</th>
    <th>등반 모임 추가하기</th>
  </tr>
  <tr>
    <td>
      <img src="https://github.com/user-attachments/assets/183b6c1f-7a88-45eb-93bc-f2e114b08937" />
    </td>
    <td>
      <img src="https://github.com/user-attachments/assets/0ea72da1-eb12-4581-96ef-a4e9c0eac80e" />
    </td>
    <td>
      <img src="https://github.com/user-attachments/assets/1845e22e-ea39-413d-ad20-d583be67d0a9" />
    </td>
  </tr>
</table>

## ☑️ 테스트 체크리스트

## 💡 집중 리뷰 요청

Button 컴포넌트에 새로운 props를 전달할 일이 생겨서 넘겼더니 `{...props}`에도 불구하고 인식하지 못했습니다. 그래서 Button 컴포넌트에 `form` prop을 추가했습니다. `{...props}`는 무용지물인걸까요?..

rwaeng committed:
🍱 채널 리스트 페이지에 사용한 아이콘 추가 (#18)

rwaeng committed:
💄 분류 추가 페이지 1차 퍼블리싱 (#21)

rwaeng committed:
💄 채널 추가 페이지 1차 퍼블리싱 (#21)

rwaeng committed:
💄 문자/전화 선택 시 입력 정보 리스트 컴포넌트 (#21)

rwaeng committed:
💄 등반 선택 시 입력 정보 리스트 컴포넌트 (#21)

hyerindev committed:
💄 모임 추가 페이지 퍼블리싱 (#21)

hyerindev committed:
🐛 최소 날짜 지정 오류 해결 (#21)

hyerindev committed:
🎨 변수명 수정 (#21)

hyerindev committed:
💄 영역을 차지하지 않는 버그 해결 (#21)

hyerindev committed:
💄 헤더 컴포넌트가 찌그러지는 버그 해결 (#21)

hyerindev committed:
💄 fieldset 간의 gap 추가 (#21)

hyerindev committed:
💄 Fieldset 컴포넌트 퍼블리싱 완성 (#21)

hyerindev committed:
💄 InputRadio 컴포넌트 퍼블리싱 완성 (#21)

hyerindev committed:
🐛 모임 유형 '문자'를 선택했을 때 모임 분류 인풋필드가 렌더링되지 않던 버그 해결 (#21)

hyerindev committed:
💄 올바르지 않은 속성 선택자 삭제 및 props로 해결 (#21)

hyerindev committed:
💄 InputText 컴포넌트 글자 수 표시 드래그 및 선택 금지 코드 추가 (#21)

hyerindev committed:
💄 글자 수 표기 위치 수정 (#21)

hyerindev committed:
💄 textarea resize handle 삭제 (#21)

hyerindev committed:
💄 InputDatepicker placeholder 삭제 (#21)

hyerindev committed:
💄 InputRadio 컴포넌트에서만 쓰는 스타일 코드 옮김 (#21)

hyerindev committed:
✨ InputDatepicker 컴포넌트 기간 지원 (#21)

hyerindev committed:
🏷️ Period 타입 코드 옮김 (#21)

hyerindev committed:
💄 NoSelect 믹스인 추가 (#21)

hyerindev committed:
🎨 텍스트 수정 (#21)

hyerindev committed:
💄 InputDropdown 컴포넌트 width가 좁아졌던 버그 해결 (#21)

hyerindev committed:
Merge branch 'dev' into 21-channelAddPage

rwaeng commented on src/components/common/InputDatepicker.tsx:
🗨️ 제가 했어야 하는 datepicker인데 혜린님이 해주셨네요,,, 감사합니다 혹시 뒤에 추가로 작업하기 싫으면 언제든 저에게 말해주세요!!!

rwaeng commented on src/components/common/InputDatepicker.tsx:
🗨️ 제가 했어야 하는 datepicker인데 혜린님이 해주셨네요,,, 감사합니다 혹시 뒤에 추가로 작업하기 싫으면 언제든 저에게 말해주세요!!!

rwaeng commented on src/components/common/InputDropdown.tsx:
💊 `cursor: pointer` 해주면 더 좋을 거 같슴니다~!!

rwaeng commented on src/pages/channel/ChannelAddPage.tsx:
❓버튼 위치가 컨텐츠 하단에 붙어서 내용이 길어지면 스크롤 내려야 보이는 걸로 생각했는데 하단에 고정하는 걸로 생각하신 거 같아서 통일이 필요할 거 같습니다!!

hyerindev replied:
기록용으로 적어둡니다. 지난 화요일 회의 결과 버튼 위치를 하단 고정으로 통일했습니다.

rwaeng commented on src/pages/channel/ChannelAddPage.tsx:
```diff
@@ -0,0 +1,127 @@
+            <InputDatepicker
+              title='등반 기간'
+              type='period'
+              min={new Date()
+                .toLocaleDateString('ko-KR', {
+                  timeZone: undefined,
+                  year: 'numeric',
+                  month: '2-digit',
+                  day: '2-digit',
+                })
+                .replace(/(\d{4}). (\d{2}). (\d{2})./, '$1-$2-$3')}
```
🗨️ 이 부분 함수로 작성해서 넣어주면 의미 전달이 더 잘될 거 같기도 해요!!

hyerindev replied:
사실 저번 프로젝트부터 몇 번이나 조언해주셔서 이번에도 코드 작성하며 아령 님의 조언을 떠올렸는데, 함수명 정하는 게 너무 어려웠던 나머지 이번에도 이렇게 적어버리고 말았습니다...

hyerindev replied:
util로 분리해서 코드 정리했습니다 😊

rwaeng reviewed (dismissed):
👍 타입 정말 꼼꼼하게 작성하셨네요 짱짱 태그 사용도 또 여럿 배워갑니다!!!

❗실행시키니까 eslint 에러 뜨던데 확인부탁드립니다~!
![image](https://github.com/user-attachments/assets/80412c40-e688-4f26-9a9e-f6deedfed95b)

🗨️ 버튼에 {...props}는 제가 타입 선언을 덜해서 안 먹었던 거 같아요!! 타입 선언 제대로 해주니까 잘 동작하네요 수정해 놓겠습니다!!

hyerindev committed:
Merge branch 'dev' into 21-channelAddPage

hyerindev commented:
> ❗실행시키니까 eslint 에러 뜨던데 확인부탁드립니다~!

집중 리뷰 요청에 이 내용을 작성하는 걸 깜빡했네요! mixin이 아직 1개만 존재해서 이 에러가 떠야 정상이에요. 꼼꼼하게 봐주셔서 감사합니다 😊

hyerindev committed:
🎨 변수명 수정 (#21)

hyerindev committed:
💄 cursor: pointer (#21)

https://github.com/BookwooriAlpineClub/Bookwoori-Front/pull/43/#discussion_r1827958232

hyerindev committed:
♻️ formatDate() util 추가 (#21)

https://github.com/BookwooriAlpineClub/Bookwoori-Front/pull/43/#discussion_r1827972736

hyerindev commented:
코드 리뷰 모두 반영했고, dev 브랜치와의 충돌을 해결했습니다. Re-request review 요청합니다!

rwaeng reviewed (dismissed):
확인했습니다~!

hyerindev committed:
Merge branch 'dev' into 21-channelAddPage
